src/manifolds/InvertibleMatrices.jl

```diff
@@ -0,0 +1,116 @@
+@doc raw"""
+    InvertibleMatrices{𝔽,T} <: AbstractDecoratorManifold{𝔽}
+
+The [`AbstractManifold`](@extref `ManifoldsBase.AbstractManifold`)  ``\operatorname{Sym}(n)``
+consisting of the real- or complex-valued invertible matrices, that is the set
+
+```math
+\bigl\{p  ∈ 𝔽^{n×n}\ \big|\ \det(p) \neq 0 \bigr\},
+```
+where the field ``𝔽 ∈ \{ ℝ, ℂ\}``.
+
+# Constructor
+
+    InvertibleMatrices(n::Int, field::AbstractNumbers=ℝ)
+
+Generate the manifold of ``n×n`` invertible matrices.
+"""
+struct InvertibleMatrices{𝔽,T} <: AbstractDecoratorManifold{𝔽}
+    size::T
+end
+
+function InvertibleMatrices(n::Int, field::AbstractNumbers=ℝ; parameter::Symbol=:type)
+    size = wrap_type_parameter(parameter, (n,))
+    return InvertibleMatrices{field,typeof(size)}(size)
+end
+
+function active_traits(f, ::InvertibleMatrices, args...)
+    return merge_traits(IsEmbeddedSubmanifold())
```

Comment:
Since in LieGroups.jl GL(n) is going to be a connection manifold, it might be a better idea to have the left-GL-invariant, right-O-invariant metric here?

Reply:
My plan was to keep this one here lightweight and do the “heavy” metric/exp/log/... stuff in LieGroups?
Otherwise, the idea to make this one a quick PR and then work on GL in LieGroups is a bit turned upside down and this here will take quite a bit of time, because next I first have to read up on the metric you mention (O is the set of orthogonal matrices for example?). And what is a connection manifold?
Since that metric Is also heavily tailored to the Lie group, it is maybe better to document it there?

Reply:
I think it would be easier to discuss this in-person.

Reply:
There is nothing that needs changing here since the metric I was talking about seems to result in an invariant exponential map.